### PR TITLE
Sidebar: Remove dead Reader code

### PR DESCRIPTION
**Pull request by kean**
Removes dead code that was under the `readerImprovements` feature flag that is long gone.

### Ghost Cells

`OldReaderPostCardCell` was still used to show "ghost" cells when Reader content is loading. It looked like the old reader card because it is what it is:

Screenshot 2024-08-29 at 12 38 21 PM

To replace it, I added a simple `ReaderGhostCell` that is simpler and better matches the current design:

Screenshot 2024-08-29 at 1 46 05 PM

Screenshot 2024-08-29 at 1 48 55 PM

I initially tried using the new reader card, but it [didn't look good](https://github.com/user-attachments/assets/1d42f0ee-53aa-44d6-a4f1-f8badd4c99ed).

I also added a simple convenience initializer to `UIStackView`:

```swift
UIStackView(axis: .vertical, alignment: .leading, spacing: 4, [
    makeLeafView(height: 10, width: 2000),
    makeLeafView(height: 10, width: .random(in: 200...600))
]),
```

I didn't want to add anything more elaborate since we are mostly using SwiftUI for new leaf views anyway.

## Undo for "Saved"

Undo for "Saved" was implemented using the following technique:

Screenshot 2024-08-29 at 10 38 42 AM

It was very complex, and we no longer use it. So it now instead shows a simple `Notice` from the bottom with an "Undo" button like everywhere else. 

I had to make this change to fully remove deprecated `ReaderPostCellActions`.

## Regression Notes

1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Modules/Sources/WordPressUI/Extensions/NSLayoutConstraint+Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    public func withPriority(_ priotity: Float) -> NSLayoutConstraint {
+        self.priority = UILayoutPriority(priotity)
+        return self
+    }
+
+    public func withPriority(_ priotity: UILayoutPriority) -> NSLayoutConstraint {
+        self.priority = priotity
+        return self
+    }
+}
```
Comment (jkmassel) [inline Modules/Sources/WordPressUI/Extensions/NSLayoutConstraint+Extensions.swift]:
```suggestion
    public func withPriority(_ priority: UILayoutPriority) -> NSLayoutConstraint {
        self.priority = priority
        return self
    }
}
```

File: Modules/Sources/WordPressUI/Extensions/NSLayoutConstraint+Extensions.swift [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import UIKit
+
+extension NSLayoutConstraint {
+    public func withPriority(_ priotity: Float) -> NSLayoutConstraint {
+        self.priority = UILayoutPriority(priotity)
+        return self
+    }
```
Comment (jkmassel) [inline Modules/Sources/WordPressUI/Extensions/NSLayoutConstraint+Extensions.swift]:
```suggestion
    public func withPriority(_ priority: Float) -> NSLayoutConstraint {
        self.priority = UILayoutPriority(priority)
        return self
    }
```

**Review by jkmassel — APPROVED** [review]
Couple of nits

**Comment by kean** [comment]
I'm going to slightly reduce the width of these secondary bars. It looks like a fence.

**Pull request merged**